From 143cebf8843187209573aa2128d51396039ec72e Mon Sep 17 00:00:00 2001
From: Stefan Berger <stefanb@linux.ibm.com>
Date: Mon, 24 Jan 2022 09:41:05 -0500
Subject: [PATCH 50/51] ima: Also allow HASH rules in IMA namespaces

Signed-off-by: Stefan Berger <stefanb@linux.ibm.com>
---
 security/integrity/ima/ima_policy.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/security/integrity/ima/ima_policy.c b/security/integrity/ima/ima_policy.c
index 7c51d3805620..7e5d7079dcba 100644
--- a/security/integrity/ima/ima_policy.c
+++ b/security/integrity/ima/ima_policy.c
@@ -1818,9 +1818,6 @@ static int ima_parse_rule(struct user_namespace *user_ns,
 		/* IMA namespace only rejects HASH rules */
 		if (ns != &init_ima_ns && result == 0) {
 			switch (entry->action) {
-			case HASH:
-				result = -EINVAL;
-				goto err_audit;
 			case AUDIT:
 				if (!capable(CAP_SYS_ADMIN)) {
 					result = -EPERM;
-- 
2.31.1

