From 3849082d2f362f27b5bfd1c153f4573da2ed647d Mon Sep 17 00:00:00 2001
From: Stefan Berger <stefanb@linux.ibm.com>
Date: Mon, 24 Jan 2022 09:53:22 -0500
Subject: [PATCH 53/55] ima: Namespace the IMA_HASH flags

---
 security/integrity/ima/ima.h | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/security/integrity/ima/ima.h b/security/integrity/ima/ima.h
index e2b0eb6a5795..4fffd6f28f4b 100644
--- a/security/integrity/ima/ima.h
+++ b/security/integrity/ima/ima.h
@@ -581,7 +581,8 @@ static inline int ima_filter_rule_match(u32 secid, u32 field, u32 op,
 #define	POLICY_FILE_FLAGS	S_IWUSR
 #endif /* CONFIG_IMA_READ_POLICY */
 
-#define IMA_NS_STATUS_ACTIONS   (IMA_AUDIT | IMA_MEASURE | IMA_APPRAISE)
+#define IMA_NS_STATUS_ACTIONS   (IMA_AUDIT | IMA_MEASURE | IMA_APPRAISE | \
+				 IMA_HASH)
 #define IMA_NS_STATUS_FLAGS     (IMA_AUDIT | IMA_AUDITED | \
 				 IMA_MEASURE | IMA_MEASURED | IMA_COLLECTED | \
 				 IMA_PERMIT_DIRECTIO | \
@@ -589,9 +590,10 @@ static inline int ima_filter_rule_match(u32 secid, u32 field, u32 op,
 				 IMA_DIGSIG_REQUIRED | \
 				 IMA_BPRM_APPRAISED | IMA_CREDS_APPRAISED | \
 				 IMA_FILE_APPRAISED | IMA_READ_APPRAISED | \
-				 IMA_MODSIG_ALLOWED | IMA_CHECK_BLACKLIST)
+				 IMA_MODSIG_ALLOWED | IMA_CHECK_BLACKLIST | \
+				 IMA_HASH | IMA_HASHED)
 
-#define IMA_IINT_FLAGS		(IMA_APPRAISE | IMA_HASH)
+#define IMA_IINT_FLAGS		0 /* FIXME: remove */
 
 static inline unsigned long iint_flags(struct integrity_iint_cache *iint,
 				       struct ns_status *ns_status)
-- 
2.31.1

